src/DataLib/ALARAJOY.cpp

```diff
@@ -0,0 +1,149 @@
+#include "ALARAJOY.h"
+#include "DataLib/ALARALib_def.h"
+#include <fstream>
+#include <sstream>
+#include <vector>
+#include <string>
+#include <cstring>
+#include <cstdio>
+
+// Define DSV row parsing structure
+std::vector<DSVRow> ALARAJOYLib::dsvData;
+
+DSVRow ALARAJOYLib::parseDSVRow(const std::string& line)
+{
+    DSVRow row;
+    std::stringstream ss(line);
+    int index;
+
+    ss >> index; 
+    ss >> row.parentKZA >> row.daughterKZA;
+    ss >> row.emittedParticles >> row.nonZeroGroups;
+
+    row.crossSections = std::vector<float>(row.nonZeroGroups);
+    for (int i = 0; i < row.nonZeroGroups; i++)
+    {
+        ss >> row.crossSections[i];
+    }
+
+    return row;
+}
+
+// Constructor
+ALARAJOYLib::ALARAJOYLib(
+    const char* transFname, const char* decayFname, const char* alaraFname
+) : EAFLib(decayFname, true),
+    currentRowIndex(0),
+    currentParent(-1)
+{
+    // Open the DSV file
+    inTrans.open(transFname, ios::in);
+    if (inTrans.is_open()) {
+        loadDSVData(); // Pre-load entire DSV
+        makeBinLib(alaraFname);
+    }
+}
+
+// Pre-load entire DSV into memory
+void ALARAJOYLib::loadDSVData()
+{
+    dsvData.clear();
+    currentRowIndex = 0;
+    currentParent = -1;
+    std::string line;
+
+    // Skip header line
+    std::getline(inTrans, line);
+
+    // Read all data rows
+    while (std::getline(inTrans, line))
+    {
+        if (line.empty()) continue; // Skip empty lines
+
+        DSVRow row = parseDSVRow(line);
+        dsvData.push_back(row);
+    }
+
+    // Set initial parent
+    if (!dsvData.empty()) {
+        currentParent = dsvData[0].parentKZA;
+    }
+}
+
+// Read library header information
+void ALARAJOYLib::getTransInfo()
+{
+
+    nParents = 0;
+
+    // Allocate arrays for maximum reactions
+    transKza = new int[MAXALARAJOYRXNS];
+    xSection = new float*[MAXALARAJOYRXNS];
+    emitted = new char*[MAXALARAJOYRXNS];
+
+    for (int rxnNum = 0; rxnNum < MAXALARAJOYRXNS; rxnNum++) {
+        xSection[rxnNum] = nullptr;
+        emitted[rxnNum]  = new char[6];
```

Comment:
magic number: can we define the maximum length of an emitted string somewhere and use that?